### PR TITLE
90753

### PR DESCRIPTION
**Pull request by saikumar9**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [90753](https://devops.dchbx.org/redmine/issues/90753) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | Philip Burrowes |

### CODE REVIEW

**Review by raghuramg — APPROVED** [review]
Code looks good. Please promote this for testing

File: app/domain/operations/hbx_enrollments/reinstate.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        current_enr = params[:hbx_enrollment]
+        # Same Employer, same Kind, same Coverage Kind and same sponsored_benefit_package_id.
+        query_criteria = {:aasm_state.nin => ['shopping', 'coverage_canceled'],
+                          :_id.ne => current_enr.id,
+                          kind: current_enr.kind,
+                          coverage_kind: current_enr.coverage_kind,
+                          sponsored_benefit_package_id: current_enr.sponsored_benefit_package_id,
+                          employee_role_id: current_enr.employee_role_id}
+        valid_enrs = current_enr.family.hbx_enrollments.where(query_criteria)
+        valid_enrs.any?{|ba| ba.effective_on >= @effective_on && ba.effective_on.year == @effective_on.year}
```
Comment (raghuramg) [inline app/domain/operations/hbx_enrollments/reinstate.rb:60]: Modified logic to use sponsored_benefit_package.effective_period to verify if any enrollment exists within the application coverage period.

**Review by raghuramg — CHANGES_REQUESTED** [review]
Please make suggested changes

File: app/domain/operations/hbx_enrollments/reinstate.rb (hunk 1 above)
Comment (saimekala07) [reply app/domain/operations/hbx_enrollments/reinstate.rb]: updated as per comments.

**Pull request merged**